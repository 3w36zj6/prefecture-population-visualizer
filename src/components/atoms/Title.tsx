import React from "react";
import styled from "styled-components";

interface TitleProps {
  text: string;
  level: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}

interface StyledHeadingProps {
  level: TitleProps["level"];
}

const fontSizeMap = {
  h1: "1.5em",
  h2: "1.4em",
  h3: "1.3em",
  h4: "1.2em",
  h5: "1.1em",
  h6: "1em",
};

const StyledHeading = styled.span<StyledHeadingProps>`
  font-size: ${({ level }) => fontSizeMap[level]};
  font-weight: bold;
`;

const Title: React.FC<TitleProps> = ({ text, level }) => {
  const Heading = level as keyof JSX.IntrinsicElements;

  return (
    <Heading>
      <StyledHeading level={level}>{text}</StyledHeading>
    </Heading>
  );
};

export default Title;
