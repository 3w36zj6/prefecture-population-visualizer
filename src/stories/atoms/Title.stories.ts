import type { Meta, StoryObj } from "@storybook/react";
import Title from "./../../components/atoms/Title";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "atoms/Title",
  component: Title,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: "centered",
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {},
  // Use `fn` to spy on the onClick arg, which will appear in the actions panel once invoked: https://storybook.js.org/docs/essentials/actions#action-args
  args: {},
} satisfies Meta<typeof Title>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Level1Title: Story = {
  args: {
    text: "Title",
    level: "h1",
  },
};

export const Level2Title: Story = {
  args: {
    text: "Title",
    level: "h2",
  },
};

export const Level3Title: Story = {
  args: {
    text: "Title",
    level: "h3",
  },
};

export const Level4Title: Story = {
  args: {
    text: "Title",
    level: "h4",
  },
};

export const Level5Title: Story = {
  args: {
    text: "Title",
    level: "h5",
  },
};

export const Level6Title: Story = {
  args: {
    text: "Title",
    level: "h6",
  },
};
