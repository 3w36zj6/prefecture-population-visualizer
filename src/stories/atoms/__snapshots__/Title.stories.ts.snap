// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Title Level1Title smoke-test 1`] = `
<h1>
  Title
</h1>
`;

exports[`atoms/Title Level2Title smoke-test 1`] = `
<h2>
  Title
</h2>
`;

exports[`atoms/Title Level3Title smoke-test 1`] = `
<h3>
  Title
</h3>
`;

exports[`atoms/Title Level4Title smoke-test 1`] = `
<h4>
  Title
</h4>
`;

exports[`atoms/Title Level5Title smoke-test 1`] = `
<h5>
  Title
</h5>
`;

exports[`atoms/Title Level6Title smoke-test 1`] = `
<h6>
  Title
</h6>
`;
